google/cloud/spanner_v1/database.py

```diff
@@ -521,6 +560,73 @@ def run_in_transaction(self, func, *args, **kw):
         finally:
             self._local.transaction_running = False
 
+    def restore(self, backup=None):
+        """Restore from a backup to this database.
+
+        :type backup: :class:`~google.cloud.spanner_v1.backup.Backup`
+        :param backup: the path of the backup being restored from.
+
+        :rtype: :class:'~google.api_core.operation.Operation`
+        :returns: a future used to poll the status of the create request
+        :raises Conflict: if the database already exists
+        :raises NotFound:
+            if the instance owning the database does not exist, or
+            if the backup being restored from does not exist
+        :raises ValueError: if backup is not set
+        """
+        if backup is None:
+            raise ValueError("Restore source not specified")
```

Comment:
Should the backup arg not have a default value if None throws a ValueError?

Reply:
The backups argument is optional because it is optional in the gapic method. The reason it is optional is because it is a oneof. If we introduce different restore sources in the future, making this optional would be a breaking change. However, I could change the argument to source instead.

Reply:
Ah ok, I missed that. I think it might be better to change it to source.

Reply:
Done.